drivers/net/slip_uart_async.c

```diff
@@ -0,0 +1,441 @@
+	uint8_t rx_buf1[UART_BUF_LEN];
+	uint8_t rx_buf2[UART_BUF_LEN];
+	uint8_t *rx_next_buf;
+
+	/* Ring buffer for incoming UART data */
+	struct ring_buf rx_ringbuf;
+	uint8_t rx_ringbuf_data[CONFIG_SLIP_UART_ASYNC_RINGBUF_SIZE];
+
+	/* TX buffer */
+	uint8_t tx_buf[UART_TX_BUF_LEN];
+	struct k_sem tx_sem;
+
+	/* Worker for processing received data */
+	struct k_work rx_work;
+	struct k_work_q rx_workq;
+	K_KERNEL_STACK_MEMBER(rx_stack, CONFIG_SLIP_UART_ASYNC_RX_STACK_SIZE);
+
+	/* MAC address */
+	uint8_t mac_addr[6];
+	struct net_linkaddr ll_addr;
+
+	/* Flags */
+	atomic_t tx_busy;
+	bool init_done;
+};
+
+static void uart_async_callback(const struct device *dev,
+				struct uart_event *evt,
+				void *user_data)
+{
+	struct slip_uart_async_context *ctx = user_data;
+	int ret;
+
+	switch (evt->type) {
+	case UART_TX_DONE:
+		LOG_DBG("TX done: %zu bytes", evt->data.tx.len);
+		atomic_clear(&ctx->tx_busy);
+		k_sem_give(&ctx->tx_sem);
+		break;
+
+	case UART_TX_ABORTED:
+		LOG_WRN("TX aborted: %zu bytes sent", evt->data.tx.len);
+		atomic_clear(&ctx->tx_busy);
+		k_sem_give(&ctx->tx_sem);
+		break;
+
+	case UART_RX_RDY:
+		/* Data received, push to ring buffer */
+		ret = ring_buf_put(&ctx->rx_ringbuf,
+				   evt->data.rx.buf + evt->data.rx.offset,
+				   evt->data.rx.len);
+		if (ret < evt->data.rx.len) {
+			LOG_WRN("RX ring buffer overflow: dropped %d bytes",
+				evt->data.rx.len - ret);
+		}
+		/* Schedule worker to process data */
+		k_work_submit_to_queue(&ctx->rx_workq, &ctx->rx_work);
+		break;
+
+	case UART_RX_BUF_REQUEST:
+		/* Provide next buffer */
+		LOG_DBG("RX buf request, next_buf=%p", ctx->rx_next_buf);
+		if (ctx->rx_next_buf) {
+			ret = uart_rx_buf_rsp(dev, ctx->rx_next_buf, UART_BUF_LEN);
+			if (ret < 0) {
+				LOG_ERR("uart_rx_buf_rsp failed: %d", ret);
+			} else {
+				LOG_DBG("Provided buffer %p", ctx->rx_next_buf);
+				/* Clear to avoid double-use */
+				ctx->rx_next_buf = NULL;
+			}
+		} else {
+			LOG_ERR("No buffer available for RX! RX will stop.");
+		}
+		break;
+
+	case UART_RX_BUF_RELEASED:
+		/* Buffer released, make it available again */
+		LOG_DBG("RX buf released: %p", evt->data.rx_buf.buf);
+		ctx->rx_next_buf = evt->data.rx_buf.buf;
+		break;
+
+	case UART_RX_DISABLED:
+		LOG_DBG("RX disabled");
+		break;
+
+	case UART_RX_STOPPED:
+		/* RX can stop due to errors (overrun/framing) during startup noise.
+		 * The UART driver will continue with the next buffer automatically.
+		 * Only log at debug level to avoid noise.
+		 */
+		LOG_DBG("RX stopped: reason %d (expected during startup)", evt->data.rx_stop.reason);
+		break;
+
+	default:
+		break;
+	}
+}
+
+static void slip_process_byte(struct slip_uart_async_context *ctx, uint8_t byte)
+{
+	int ret;
+	static int scan_count = 0;
+
+	switch (ctx->rx_state) {
+	case STATE_SCANNING_FOR_FRAME:
+		scan_count++;
+		if (byte == COBS_DELIMITER) {
+			/* Start of frame */
+			LOG_INF("✓ Found COBS delimiter (0x00), starting frame (scanned %d bytes)", scan_count);
```

Comment:
I am not sure if it is a good idea to print non ascii chars here, it does not give much benefit and might cause some garbage to be printed.
Also this looks like a debug print than an info message. Usually there is no need to print this kind of messages unless you are debugging things.

Reply:
I think this is gone now.